### PR TITLE
Update questionnaire buttons in xml fragment

### LINKED ISSUE #2353: Questionnaire content style customization
**Is your feature request related to a problem? Please describe.**
It's not possible to customize the padding for the RecyclerView that contains the questionnaire content.

**Describe the solution you'd like**
FHIRCore has a UX design update that requires modifying the padding around the questionnaire content. To achieve this, the proposal is to add a style to the RecyclerView that will allow for customization.

**Describe alternatives you've considered**

**Additional context**
This is related to the UX Questionnaire adjustments on FHIRCore [Styling update to forms](https://github.com/opensrp/fhircore/issues/2910)

Updated design
questionnaire_content_padding

questionnaire_button_padding

questionnaire_button_text_padding

**Would you like to work on the issue?**
Yes.

### PR DESCRIPTION
**Pull request by sharon2719**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #2353 

**Description**   
The `submit` button has been made configurable and the layout can be changed based on one's use case. The rest of the buttons `previous`, `next` and `review` have also been made configurable as well based on the designs in the issue above.  

Clear and concise code change description. 

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one:Feature

**Screenshots (if applicable)**
Screenshot from 2023-12-22 12-40-38

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

File: datacapture/src/main/res/values/styles.xml [hunk 1]
```diff
@@ -107,7 +107,7 @@
         <item
             name="questionnaireChipStyle"
         >@style/Widget.Material3.Chip.Input</item>
-        <item name="questionnaireButtonStyle">@style/Widget.Material3.Button
+        <item name="questionnaireButtonStyle">@style/Questionnaire.ButtonStyle
```
Comment (MJ1998) [inline datacapture/src/main/res/values/styles.xml:110]:
```suggestion
        <item name="questionnaireButtonStyle">Questionnaire.ButtonStyle
```

With style it gives me error: Cannot resolve symbol '@style/Questionnaire.ButtonStyle'

**Review by MJ1998 — CHANGES_REQUESTED** [review]

**Comment by MJ1998** [comment]
Thanks. Please add a description to the solution!

**Comment by sharon2719** [comment]
@MJ1998  I did further testing and noted that if the `@style` is not added then it fails with this error `Android resource linking failed
com.google.android.fhir.demo-mergeDebugResources-60:/values/values.xml:7034: error: expected reference but got (raw string) Questionnaire.ButtonStyle.
error: failed linking references.`. However when its added it works just fine

**Comment by MJ1998** [comment]
@sharon2719 

You are right! My IDE was throwing some error. After invalidate-restart its good.

**Review by MJ1998 — APPROVED** [review]
LGTM

**Comment by sharon2719** [comment]
@MJ1998 Kindly prompt kokoro

File: datacapture/src/main/res/values/styles.xml (hunk 1 above)
Comment (jingtang10) [reply datacapture/src/main/res/values/styles.xml:110]: @sharon2719 did you address this comment?

File: datacapture/src/main/res/values/styles.xml [hunk 2]
```diff
@@ -397,13 +397,25 @@
...
+        <item name= "android:layout_width">wrap_content </item>
+        <item name= "android:layout_height">wrap_content </item>
     </style>
     <style
         name="Questionnaire.CancelButtonStyle"
         parent="Widget.Material3.Button.TextButton"
     >
         <item name="android:textAllCaps">false</item>
         <item name="android:visibility">visible</item>
+        <item name= "android:layout_width">wrap_content </item>
```
Comment (jingtang10) [inline datacapture/src/main/res/values/styles.xml:409]:
```suggestion
        <item name= "android:layout_width">wrap_content</item>
```

File: datacapture/src/main/res/values/styles.xml [hunk 3]
```diff
@@ -397,13 +397,25 @@
...
+        <item name= "android:layout_height">wrap_content </item>
```
Comment (jingtang10) [inline datacapture/src/main/res/values/styles.xml:410]:
```suggestion
        <item name= "android:layout_height">wrap_content</item>
```

File: datacapture/src/main/res/values/styles.xml [hunk 4]
```diff
@@ -397,13 +397,25 @@
...
+    </style>
+
+    <style
+        name="Questionnaire.ButtonStyle"
+        parent="Widget.Material3.Button"
+        >
+        <item name= "android:layout_width">wrap_content </item>
```
Comment (jingtang10) [inline datacapture/src/main/res/values/styles.xml:417]:
```suggestion
        <item name= "android:layout_width">wrap_content</item>
```

File: datacapture/src/main/res/values/styles.xml [hunk 5]
```diff
@@ -397,13 +397,25 @@
...
+        <item name= "android:layout_height">wrap_content </item>
```
Comment (jingtang10) [inline datacapture/src/main/res/values/styles.xml:418]:
```suggestion
        <item name= "android:layout_height">wrap_content</item>
```

File: datacapture/src/main/res/values/styles.xml (hunk 1 above)
Comment (sharon2719) [reply datacapture/src/main/res/values/styles.xml:110]: @jingtang10 Yes this was addressed. @MJ1998 had an issue with his machine while testing but after an update it was sorted and working as expected

**Review by ellykits — APPROVED** [review]
LGTM

File: datacapture/src/main/res/values/styles.xml [hunk 6]
```diff
@@ -396,12 +396,26 @@
...
+        <item name="android:visibility">visible</item>
+        <item name= "android:layout_width">wrap_content</item>
+        <item name= "android:layout_height">wrap_content</item>
     </style>
     <style
         name="Questionnaire.CancelButtonStyle"
         parent="Widget.Material3.Button.TextButton"
     >
         <item name="android:textAllCaps">false</item>
+        <item name="android:visibility">visible</item>
```
Comment (santosh-pingle) [inline datacapture/src/main/res/values/styles.xml:408]:
```suggestion
```

**Review by santosh-pingle — DISMISSED** [review]
I think we can remove the visibility attribute change, as it get handled through the flags in the configuration.

**Comment by MJ1998** [comment]
Build fails on spotless apply.
Can you run spotless and raise a commit ?
Thanks @sharon2719

**Pull request merged**